src/module.c

```diff
@@ -2254,6 +2254,34 @@ int moduleIsModuleCommand(void *module_handle, struct serverCommand *cmd) {
     return (cp->module == module_handle);
 }
 
+/* ValkeyModule_UpdateRuntimeArgs can be used to update the module argument values.
+ * The function parameter 'argc' indicates the number of updated arguments, and 'argv'
+ * represents the values of the updated arguments.
+ * Once 'CONFIG REWRITE' command is called, the updated argument values can be saved into conf file.
+ * One example can be found in file tests/modules/moduleparameter.c.
+ *
+ * Returns:
+ * - VALKEYMODULE_OK on successfully updating.
+ * - VALKEYMODULE_ERR on failure.
+ */
+int VM_UpdateRuntimeArgs(ValkeyModuleCtx *ctx, ValkeyModuleString **argv, int argc) {
+    if (!ctx->module->onload) {
+        return VALKEYMODULE_ERR;
+    }
```

Comment:
Why is this only allowed at startup? I thought the whole point was they were able to update these at any point in time.

Reply:
Good point. I'm surprised the test case passes.

Reply:
The condition "!ctx->module->onload" is checked here is not only allowed to update at startup, the goal is to check if ctx->module->onload is NULL to prevent NULL pointer exception. This API can be called at any time during runtime.

Reply:
I think it means that onload is ongoing. If it doesn't, then there is something wrong.

I tried to follow the function calls that happen when a module is loaded. This is what I could find:

* MODULE LOAD -> moduleCommand -> moduleLoad.
* moduleLoad loads the module file using dlopen.
* moduleLoad finds the ValkeyModule_OnLoad function in the module using dlsym.
* moduleLoad calls the ValkeyModule_OnLoad in the module.
* The module's ValkeyModule_OnLoad function calls ValkeyModule_Init.
* ValkeyModule_Init (implemented in valkeymodule.h) calls ValkeyModule_SetModuleAttribs.
* VM_SetModuleAttribs creates the struct ctx->module and sets ctx->module->onload = 1.
* The ValkeyModule_OnLoad function calls other functions, such as ValkeyModule_CreateCommand
* The ValkeyModule_OnLoad function returns. 
* moduleLoad logs serverLog(LL_NOTICE, "Module '%s' loaded from %s", ctx.module->name, path);
* moduleLoad sets ctx.module->onload = 0;
* This ctx is freed, but the module object is added to a dict.

There is no other place that onload is set. That means onload = 1 only when ValkeyModule_OnLoad is running.

Reply:
The onload field in ValkeyModule is not a pointer. It is an int.

    int onload;                           /* Flag to identify if the call is being made from Onload (0 or 1) */

Reply:
Sorry, it looks like on that time, my brain is offline.  Yes, you are right, it is int, not pointer.  I add this condition here to prevent this API is called if module loaded fail